### ISSUE TITLE
Missing package.json file

### ISSUE THREAD

**Issue by NickGlenn** [MarkTA/ModalTree#2]
I can't install the project and test it because no `package.json` file has been added.

**Comment by MarkTA** [comment]
OK I just created a new directory and new files on my computer and copy pasted the code. Can I just copy and past the old json file over to the new directory or do I need to init something into the folder?

**Comment by NickGlenn** [comment]
Yes, just copy the package.json file into the new directory and then add/commit the file like so...

```
git add package.json
git commit -a -m "Added package.json file"
git push origin master
```

**Comment by MarkTA** [comment]
Tried to copy paste the file into the new directory and got this error-

$ git push origin master

To https://github.com/MarkTA/ModalTree.git

 ! [rejected]        master -> master (non-fast-forward)

error: failed to push some refs to 'https://github.com/MarkTA/ModalTree.git'

hint: Updates were rejected because a pushed branch tip is behind its remote

hint: counterpart. Check out this branch and integrate the remote changes

hint: (e.g. 'git pull ...') before pushing again.

**Comment by MarkTA** [comment]
This is a new directory with new files. Do I need to npm install something into my directory folder to create a unique package.json?

**Comment by NickGlenn** [comment]
Did you accidentally copy the `.git` folder over by chance?  The `.git` folder contains all history information for your local copy of the repository.

You should just be able to copy the old `package.json` file from your original repo and added that using the commands I mentioned above _unless_ your `.git` folder got corrupted.

If the `.git` folder is borked, you can always clone a new copy of the un-corrupted remote repository using the following command:

```
git clone git@github.com:MarkTA/ModalTree.git
```

You can also add a desired folder at the end to put the contents into.

```
git clone git@github.com:MarkTA/ModalTree.git ./ModalTree-Project
```

**Comment by MarkTA** [comment]
no all that I have in my new directory is the index.html, app.css, and app.js

**Comment by MarkTA** [comment]
aside from the new git folder

**Comment by MarkTA** [comment]
I didn't copy paste the files, I created new files and copy pasted the content

**Comment by MarkTA** [comment]
I don't have express, or any other node packages installed in my new directory.

**Comment by NickGlenn** [comment]
Hmmm…


So copy the `package.json` file into the new directory and add the following contents in a `.gitignore` file.




```

node_modules

```




Then try the following commands:




```

cd {your new project directory}

git add .

git commit -a -m “Added package.json and .gitignore file”

npm install

node server.js

```




And let me know if the server boots up.


Nick Glenn

**Comment by NickGlenn** [comment]
Hmmm…

So copy the `package.json` file into the new directory and add the following contents in a `.gitignore` file.

```
node_modules
```

Then try the following commands:

```
cd {your new project directory}
git add .
git commit -a -m “Added package.json and .gitignore file”
npm install
node server.js
```

And let me know if the server boots up.

**Comment by MarkTA** [comment]
OK I was able to commit and push the file with github desktop. I think maybe that has been the problem, conflict between github desktop and the terminal?

**Comment by NickGlenn** [comment]
It shouldn’t because the GitHub GUI simply reflects the .git folder but it’s possible?

Nick Glenn

**Comment by MarkTA** [comment]
mark at Marks-MBP in ~/documents/ModalTree (MarksBranch) 
$ npm install
npm WARN package.json mark@1.0.0 No description
npm WARN package.json mark@1.0.0 No repository field.
npm WARN package.json mark@1.0.0 No README data

mark at Marks-MBP in ~/documents/ModalTree (MarksBranch) 
$ node server.js
module.js:338
    throw err;
          ^
Error: Cannot find module '/Users/mark/Documents/ModalTree/server.js'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

**Comment by MarkTA** [comment]
Just realized that I wasn't in the master branch while npm installing. Is that an issue?

**Comment by NickGlenn** [comment]
Shouldn't be because the `node_modules` file is being ignored by git.

**Comment by MarkTA** [comment]
I'm going to focus on other things right now. Can we meet for a few minutes tonight to fix this?

**Comment by NickGlenn** [comment]
Sure thing.


Nick Glenn
